pkg/probe/turbo_probe.go

```diff
@@ -155,6 +143,24 @@ func (theProbe *TurboProbe) ValidateTarget(accountValues []*proto.AccountValue)
 	return validationResponse
 }
 
+// ========================================== Action Processing =======================================================
+
+func (theProbe *TurboProbe) ExecuteAction(actionExecutionDTO *proto.ActionExecutionDTO, accountValues []*proto.AccountValue,
+	progressTracker ActionProgressTracker) *proto.ActionResult {
+	if theProbe.ActionClient == nil {
+		glog.Infof("ActionClient not defined for Probe ", theProbe.ProbeType)
+		return theProbe.createActionErrorDTO("ActionClient not defined for Probe " + theProbe.ProbeType)
+	}
+	glog.Infof("Execute Action for Target : ", accountValues)
+	response := theProbe.ActionClient.ExecuteAction(actionExecutionDTO, accountValues, progressTracker)
+
+	if response == nil {
```

Comment:
let ActionClient.ExecuteAction return an error so that we check
```go
response, err := theProbe.ActionClient.ExecuteAction(actionExecutionDTO, accountValues, progressTracker)
if err != nil {
  // Handle error case.
}
```